backend/src/controllers/RequestController.integration.test.ts

```diff
@@ -0,0 +1,496 @@
+      expect(response.body).toHaveProperty("errMsg");
+    });
+
+    it("should return NOT_MODIFIED if the request could not be approved", async () => {
+      const approvalDetails = {
+        performedBy: 140001,
+        requestId: 99999,
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/approveRequest")
+        .send(approvalDetails);
+
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe("rejectRequest", () => {
+    it("should reject the request for valid rejection details", async () => {
+      const rejectionDetails = {
+        performedBy: 140001,
+        requestId: 12345,
+        reason: "Not required anymore",
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/rejectRequest")
+        .send(rejectionDetails);
+
+      expect(response.status).toBe(200);
+    });
+
+    it("should return an error if rejection details are invalid", async () => {
+      const invalidRejectionDetails = {
+        performedBy: "invalid-id",
+        requestId: null,
+        reason: "",
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/rejectRequest")
+        .send(invalidRejectionDetails);
+
+      expect(response.status).toBe(200);
+    });
+
+    it("should return NOT_MODIFIED if the request could not be rejected", async () => {
+      const rejectionDetails = {
+        performedBy: 140001,
+        requestId: 99999,
+        reason: "Request is invalid",
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/rejectRequest")
+        .send(rejectionDetails);
+
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe("revokeRequest", () => {
+    it("should revoke the request for valid revocation details", async () => {
+      const revocationDetails = {
+        performedBy: 140001,
+        requestId: 12345,
+        reason: "No longer needed",
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/revokeRequest")
+        .send(revocationDetails);
+
+      expect(response.status).toBe(200);
+    });
+
+    it("should return an error if revocation details are invalid", async () => {
+      const invalidRevocationDetails = {
+        performedBy: "invalid-id",
+        requestId: null,
+        reason: "",
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/revokeRequest")
+        .send(invalidRevocationDetails);
+
+      expect(response.status).toBe(200);
+    });
+
+    it("should return NOT_MODIFIED if the request could not be revoked", async () => {
+      const revocationDetails = {
+        performedBy: 140001,
+        requestId: 99999,
+        reason: "Request is invalid",
+      };
+
+      const response = await request(mockServer)
+        .post("/api/v1/revokeRequest")
+        .send(revocationDetails);
+
+      expect(response.status).toBe(200);
+    });
+  });
```

Comment:
need to check whats the response, missing       expect(response.text).toEqual(HttpStatusResponse.NOT_MODIFIED);